File: 2_algorithms/grokking-algorithms/4_selection_sort/python/selection_sort.py
def min_element(l) -> int:

    min_element = l[0]
    min_index = 0

    for i in range(1, len(l)):

        if l[i] < min_element:

            min_element = l[i]
            min_index = i

    return min_index

def selection_sort(l) -> list:

    new_list = []

    for i in range(len(l)):

        min_index = min_element(l)

        new_list.append(l.pop(min_index))
    
    return new_list


print(selection_sort([5, 3, 6, 2, 10]))

File: 2_algorithms/grokking-algorithms/4_selection_sort/go/selection_sort.go
package selectionsort

import "fmt"

func getMin(arr []int) int {

	var minValue = arr[0]
	var minIndex = 0

	for i, e := range arr {

		if e < minValue {
			minValue = e
			minIndex = i
		}
	}

	return minIndex
}

func selectionSort(arr []int) []int {

	var (
		newArr []int
	)

	for len(arr) > 0 {
		minIndex := getMin(arr)
		newArr = append(newArr, arr[minIndex])
		arr = append(arr[:minIndex], arr[minIndex+1:]...)
	}

	return newArr
}

func SelectionSort() {
	fmt.Println(selectionSort([]int{5, 3, 6, 2, 10}))
}
